client/store/reducers/chapter.ts

```diff
@@ -0,0 +1,39 @@
+import { chapterTypes } from '../types';
+import produce from 'immer';
+
+const initialState: chapterTypes.IChapterStoreState = {
+  name: '',
+  description: '',
+  category: '',
+  loading: false,
+  error: '',
+};
+
+const reducer = (
+  state = initialState,
+  action: chapterTypes.IChapterActionTypes,
+) =>
+  produce(state, draft => {
```

Comment:
Instead of using immer, why not just.
```
const reducer = (state = initialState, action: IChapterActionTypes) => {
  switch (action.type) {
    case FETCH_START:
      return {
        ...state,
        error: '',
        loading: true
      }
    case FETCH_SUCCESS:
      return {
        ...state,
        ...action.payload,
        error: '',
        loading: false
      }
    case FETCH_FAIL:
      return {
        ...state,
        error: action.error,
        loading: false
      }
    default:
      return state;
  }
}
```

Reply:
Vanilla JS destructuring works only with one level nested JS Objects, and can get pretty complicated to maintain when API responses get a little bit complicated (using nested JSON Objects). This is the biggest `WHY` that I like to use `immerjs`. More benefits can be found [here](https://immerjs.github.io/immer/docs/introduction#benefits).


__Why not `ImmutableJS`__

ImmutableJS uses it's own set of data structures, which for new comers or even with core JS devs might become a barrier to contribute. Where as `ImmerJS` keeps everything similar to pure js, but also giving us benefits of Object freezing, easy immutability with nested objects and more.

Reply:
How would one enforce immutability of state via. native JS without tons of boilerplate?  @Zeko369

Reply:
@Shub1427 I agree with you that this is better for more complex stores

Reply:
Will resolve this @Zeko369